TFT/src/User/Menu/Popup.c

```diff
@@ -195,7 +195,7 @@ void _setDialogCancelTextLabel(int16_t index)
 */
 void showDialog(DIALOG_TYPE type, void (*ok_action)(), void (*cancel_action)(), void (*loop_action)())
 {
-  if (!(infoSettings.mode & 1) == MODE_MARLIN)  // if standard/blocked Marlin mode, then exit
+  if ((infoSettings.mode & 1) == MODE_MARLIN)  // if standard/blocked Marlin mode, then exit
```

Comment:
The `(infoSettings.mode & 1)` is not needed. The value of **infoSettings.mode** is always assigned absolute values. So performing the bitwise is overkill for this purpose. `(infoSettings.mode == MODE_MARLIN)` will give the same result with 1 less CPU cycle.

Reply:
no because default_mode can now assume values from 0-3. 0 and 2 are for Marlin mode while 1 and 3 are for Touch mode. Same logic you used for fil_runout. The bitwise operator is the faster we can use. As mentioned by Kisslorand (on another thread)  a possible optimization could be applied to the logic to alternate the values for boolean value. B = (B + 1) % 2 could be converted to a more optimized B = 1 - B (it's ok for both boolean and integer that can assume values in range 0-1)

Reply:
> no because default_mode can now assume values from 0-3. 0 and 2 are for Marlin mode while 1 and 3 are for Touch mode.

If this is the case then, IMHO, I think the `LCD_MODE` in Settings.h should be updated to include the 4 values because it is currently providing incomplete values.

Using logical operators like % make the code easier to understand. The C/C++ compiler automatically converts the logical operators like % to bitwise operators, so it does not make sense to change all the existing code for this purpose unless the code is specifically changing each bit individually like the Filament runout and like in parameter settings. 
**(B + 1) % 2** is used to give possibility to change the value range. **B = 1 - B** is unsafe for unsigned integers. Instead of **B = 1 - B** it is better to use **B = !B** to toggle boolean values which is also safe against negative values and faster than **B = 1 - B**.

Reply:
for infoSettings.mode I defined MAX_MODE_COUNT 4 (used in config.c). However I could also extend LCD_MODE including the blocked Marlin/Touch modes. But I should also make small changes in the function HW_InitMode (that currently simply needs to know if Marlin or Touch mode has to be configured. blocked Marlin/Touch are equivalent to Marlin/Touch mode).

I know that for boolean type the standard is B = !B but it is equivalent to "if B != 0 ? B = true : B = false)" so it should be less compact and faster. If we guarantee (as it is already now) that initial value for B is 0 or 1 (in config.ini and parsed by config.c), then B = 1 - B is safe and it should be more compact and faster than B = !B. However, it was just an opinion for a  possible optimization. Sure, it's not a must to implement